### DISCUSSION TITLE
Is there any different between run FlareSolverr docker at local machine and on the cloud?

### DISCUSSION THREAD

**Discussion by mnhlt** [FlareSolverr/FlareSolverr#1395] [Q&A]
Have a good day everyone!

I have trial which combines Puppeteer and FlareSolverr. The setup could be described by following step:
1. Start FlareSolverr with docker. -> api url: http://localhost:8191/v1
2. Send a get command with website <url> having challenge to Flare Solverr -> solved and return **cookies + user agent**.
3. Set **cookies + user agent** to puppeteer page.
4. use puppeteer page to handle website afterward.

The setup works seamless at my local machine.

But the problem come when I try to deploy FlareSolverr to cloud and provide it with domain such as crawler.com -> api url: https://crawler.com/v1. at this point, cookies and user agent seem does not work with puppeteer anymore and I get challenge every time.

Just wonder if there is any different between setups?

**Comment by mnhlt** [comment]
Update:
I figured out that if i run puppeteer on the cloud machine, I work as well.
So the setup only if the puppeteer and FlareSolverr are on the same machine.

Could we discuss on how it works like that?

**Comment by ilike2burnthing** [comment]
https://github.com/FlareSolverr/FlareSolverr/wiki/Troubleshooting
